File: configor/src/main/java/com/bruceewu/configor/entity/NormalSchemaor.java
package com.bruceewu.configor.entity;

import android.text.TextUtils;

public class NormalSchemaor implements ISchemaor {
    private String title;
    public String subTitle;
    public String img;
    public String action;
    public String url;
    public int login;

    public static NormalSchemaor newInstance(String title, String subTitle, String img, String action, String url, int login) {
        NormalSchemaor schemaor = new NormalSchemaor();
        schemaor.title = title;
        schemaor.subTitle = subTitle;
        schemaor.img = img;
        schemaor.action = action;
        schemaor.url = url;
        schemaor.login = login;
        return schemaor;
    }

    @Override
    public long getId() {
        return 0;
    }

    @Override
    public String getName() {
        return null;
    }

    @Override
    public String getLmNumber() {
        return null;
    }

    @Override
    public String getAdId() {
        return null;
    }

    @Override
    public boolean isNative() {
        return TextUtils.equals(action, "native");
    }

    @Override
    public boolean needLogin() {
        return login == 1;
    }

    @Override
    public boolean isCredit() {
        return false;
    }

    @Override
    public String getTitle() {
        return title;
    }

    @Override
    public String getLink() {
        return url;
    }
}

File: configor/src/main/java/com/bruceewu/configor/entity/NormalSchemaor.kt
package com.bruceewu.configor.entity

import android.text.TextUtils

//常用的跳转逻辑
open class NormalSchemaor : ISchemaor {
    private var title: String = ""
    var subTitle: String = ""
    var img: String = ""
    var action: String = ""
    var url: String = ""
    var login: Int = 0

    companion object {
        fun newInstance(title: String?, subTitle: String?, img: String?, action: String?, url: String?, login: Int?): NormalSchemaor = NormalSchemaor().apply {
            this.title = title ?: ""
            this.subTitle = subTitle ?: ""
            this.img = img ?: ""
            this.action = action ?: ""
            this.url = url ?: ""
            this.login = login ?: 0
        }
    }

    override fun getId() = 0L

    override fun getName() = ""

    override fun getLmNumber() = ""

    override fun getAdId() = ""

    override fun isNative() = TextUtils.equals(action, "native")

    override fun needLogin() = login == 1

    override fun isCredit() = false

    override fun getTitle() = title

    override fun getLink() = url
}

//跳转H5页面
class H5Schemaor(val url: String?) : ISchemaor {

    override fun getLink() = url

    override fun getName() = ""

    override fun getId() = 0L

    override fun getLmNumber() = ""

    override fun isNative() = false

    override fun getTitle() = ""

    override fun needLogin() = true

    override fun isCredit() = false

    override fun getAdId() = ""
}

//跳转本地页面
class NativeSchemaor(val url: String) : ISchemaor {
    override fun getLink() = url

    override fun getName() = ""

    override fun getId() = 0L

    override fun getLmNumber() = ""

    override fun isNative() = true

    override fun getTitle() = ""

    override fun needLogin() = true

    override fun isCredit() = false

    override fun getAdId() = ""
}

//当前页面直接交互操作
class ActionSchemaor(val url: String?) : ISchemaor {
    override fun getId() = 0L

    override fun getName() = ""

    override fun getLmNumber() = ""

    override fun getAdId() = ""

    override fun isNative() = true

    override fun needLogin() = true

    override fun isCredit() = false

    override fun getTitle() = ""

    override fun getLink() = url
}

class NoneSchemaor : ISchemaor {
    override fun getLink() = ""

    override fun getName() = ""

    override fun getId() = 0L

    override fun getLmNumber() = ""

    override fun isNative() = true

    override fun getTitle() = ""

    override fun needLogin() = false

    override fun isCredit() = false

    override fun getAdId() = ""
}

class NotificationSchemaor(private val isH5: String, private val url: String, private val needLogin: String) : ISchemaor {

    override fun getLink() = url

    override fun getName() = ""

    override fun getId() = 0L

    override fun getLmNumber() = ""

    override fun isNative() = isH5 != "1"

    override fun getTitle() = ""

    override fun needLogin() = needLogin == "1"

    override fun isCredit() = false

    override fun getAdId() = ""
}